### PR TITLE
feature: add tool to configure operations and configure AAVE for Base

### PR DESCRIPTION
**Pull request by robercano**
Added a tool to check configurations from `OperationRegistry` and to add it as well. It supports indicating which part of the config to add, for example `aave.v3` will only add the operations for AAVE v3.

Also added entry to the Developer Handbook: https://www.notion.so/oazo/Operation-Registry-Tool-Beta-b8cdb22edde84966a6a04dbf9071de48

Additionally fixed some leftovers from the Base deployment

### CODE REVIEW

File: packages/dma-contracts/tasks/common/generic-utils.ts [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+ *
+ * The path 'a.b.c' will return 1
+ *
+ * @param obj The object to extract the property from
+ * @param path The path to the property in string format
+ *
+ * @returns The property value
+ */
+export function getPropertyFromPath(obj: any, path: string): any {
+  const pathParts = path.split('.')
```
Comment (halaprix) [inline packages/dma-contracts/tasks/common/generic-utils.ts:20]: nice :)

File: packages/dma-contracts/tasks/common/verification-utils.ts [hunk 2]
```diff
@@ -100,3 +115,90 @@ export class ActionsDatabase {
     })
   }
 }
+
+export function validateActionHashes(
+  operationHashes: string[],
+  operationOptionals: boolean[],
```
Comment (halaprix) [inline packages/dma-contracts/tasks/common/verification-utils.ts]: maybe `optionalOperationActions` ?

File: packages/dma-contracts/tasks/common/verification-utils.ts [hunk 3]
```diff
@@ -100,3 +115,90 @@ export class ActionsDatabase {
     })
   }
 }
+
+export function validateActionHashes(
+  operationHashes: string[],
```
Comment (halaprix) [inline packages/dma-contracts/tasks/common/verification-utils.ts]: `operationActionsHashes` ? I'm suggesting as I'm not familiar with that part of the codebase so it wasnt all obvious at first glance

File: packages/dma-contracts/tasks/common/verification-utils.ts [hunk 4]
```diff
@@ -100,3 +115,90 @@ export class ActionsDatabase {
...
+      actionDefinitions[actionIndex].optional !== operationOptionals[actionIndex]
+    ) {
+      actionsValidated = false
+
+      const actionDefinitionName = actionsDatabase.getActionName(
+        actionDefinitions[actionIndex].hash,
+      )
+      const operationHashName = actionsDatabase.getActionName(operationHashes[actionIndex])
+
+      actionsErrorMessage = `Action ${actionIndex} expected hash ${
```
Comment (halaprix) [inline packages/dma-contracts/tasks/common/verification-utils.ts]: maybe aggregate the error message if more actions are invalid so you dont have to fix , run again, etc? Or thats unlikely?

File: packages/dma-contracts/tasks/common/verification-utils.ts (hunk 2 above)
Comment (halaprix) [reply packages/dma-contracts/tasks/common/verification-utils.ts]: or `operationOptionalActions`

File: packages/dma-contracts/tasks/common/verification-utils.ts [hunk 5]
```diff
@@ -23,18 +25,31 @@ export type ActionValidationResult = {
...
 }
 
-export type VerificationResult = {
+export type ValidationResult = {
   success: boolean
   totalEntries: number
   totalValidated: number
 }
 
+export enum OpValidationResultType {
```
Comment (halaprix) [inline packages/dma-contracts/tasks/common/verification-utils.ts:46]: I'd expand the comments to be more ELI5 , so frst time visitor know what is what. Eg the registry is an onchain registry .

Or maybe thats enough, to precise that the registry is an onchain registry of the operations

File: packages/dma-contracts/tasks/operations-registry/index.ts [hunk 6]
```diff
@@ -0,0 +1,212 @@
...
+      }
+
+      const operationHashes: string[] = operationDefinition.actions.map(
+        (action: ActionDefinition) => action.hash,
+      )
+      const operationOptionals: boolean[] = operationDefinition.actions.map(
+        (action: ActionDefinition) => action.optional,
+      )
+
+      const tx = await operationRegistry.addOperation({
```
Comment (halaprix) [inline packages/dma-contracts/tasks/operations-registry/index.ts:163]: its not behind multisig ?

File: packages/dma-contracts/tasks/service-registry/index.ts [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+      serviceAddress = await serviceRegistry.getRegisteredService(service.serviceRegistryName)
+    } catch (e) {
+      console.log(`[❌][FETCH ERROR]  ${service.name}: ${e}`)
+    }
+
+    if (isInvalidAddress(serviceAddress)) {
+      const serviceNameHash = await serviceRegistry.getServiceNameHash(service.serviceRegistryName)
+
+      try {
+        const tx = await serviceRegistry.addNamedService(serviceNameHash, service.address)
```
Comment (halaprix) [inline packages/dma-contracts/tasks/service-registry/index.ts:126]: what about multisig ?

File: packages/dma-contracts/tasks/service-registry/index.ts [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+  console.log(`Network: ${network}`)
+  console.log(`Path: ${configPath}`)
+  console.log(`ServiceRegistry: ${serviceRegistry.address}`)
+  console.log('==============================\n')
+
+  for (const serviceName of Object.keys(property)) {
+    const service = property[serviceName] as ConfigEntry
+
+    if (!service.serviceRegistryName) {
+      console.log(`[✅][LOCAL CONFIG] ${service.name}: ${service.address}`)
```
Comment (halaprix) [inline packages/dma-contracts/tasks/service-registry/index.ts]: maybe indicate here that its `not required in SR` or something similar. `LOCAL CONFIG` got me confused so I had to cehck what does taht mean :)

**Review by halaprix — CHANGES_REQUESTED** [review]

File: packages/dma-contracts/tasks/common/verification-utils.ts (hunk 5 above)
Comment (robercano) [reply packages/dma-contracts/tasks/common/verification-utils.ts:46]: I tried to explain it better, let me know if you like it more now

File: packages/dma-contracts/tasks/common/verification-utils.ts (hunk 3 above)
Comment (robercano) [reply packages/dma-contracts/tasks/common/verification-utils.ts]: 👍

File: packages/dma-contracts/tasks/common/verification-utils.ts (hunk 2 above)
Comment (robercano) [reply packages/dma-contracts/tasks/common/verification-utils.ts]: I've gone with `operationIsActionOptional` because for me the other names imply that it is a list of optional actions, when in reality is a list of whether the corresponding action is optional. Let me know if you like it!

File: packages/dma-contracts/tasks/common/verification-utils.ts (hunk 4 above)
Comment (robercano) [reply packages/dma-contracts/tasks/common/verification-utils.ts]: I've tried something here to aggregate the message, let me know what you think

File: packages/dma-contracts/tasks/operations-registry/index.ts (hunk 6 above)
Comment (robercano) [reply packages/dma-contracts/tasks/operations-registry/index.ts:163]: Yes, that is why there is a disclaimer in the Notion page about this tool being only valid for an initial deployment, when the deployer is still the owner. For the other functionality, we will need to integrate the Safe SDK, but we can definitely reuse this tools and add a flag to indicate whether we want to add the operations, or propose them in Safe

File: packages/dma-contracts/tasks/service-registry/index.ts (hunk 8 above)
Comment (robercano) [reply packages/dma-contracts/tasks/service-registry/index.ts]: This is a tricky one to summarize in the label. But I've added `NOT REQUIRED`, do you think it is better?

File: packages/dma-contracts/tasks/service-registry/index.ts (hunk 7 above)
Comment (robercano) [reply packages/dma-contracts/tasks/service-registry/index.ts:126]: Same answer as above

File: packages/dma-contracts/tasks/common/verification-utils.ts (hunk 4 above)
Comment (halaprix) [reply packages/dma-contracts/tasks/common/verification-utils.ts]: nice :)

File: packages/dma-contracts/tasks/common/verification-utils.ts [hunk 9]
```diff
@@ -23,18 +25,60 @@ export type ActionValidationResult = {
...
+export type ValidationResult = {
   success: boolean
   totalEntries: number
   totalValidated: number
 }
 
+/**
+ * Result of validating the state of the OperationRegistry. The registry is an onchain
+ * contract that holds information about which operations are allowed in the system. Each
+ * operation is a set of actions, defined by their name's hashes, pluse a list of booleans
```
Comment (halaprix) [inline packages/dma-contracts/tasks/common/verification-utils.ts]:
```suggestion
 * operation is a set of actions, defined by their name's hashes, plus a list of booleans
```

File: packages/dma-contracts/tasks/operations-registry/index.ts (hunk 6 above)
Comment (halaprix) [reply packages/dma-contracts/tasks/operations-registry/index.ts:163]: ah ok ! I haven't read the notion page :)

**Review by halaprix — APPROVED** [review]
